src/PlusDataCollection/OpenHaptics/vtkPlusOpenHapticsDevice.cxx

```diff
@@ -0,0 +1,294 @@
+vtkPlusOpenHapticsDevice::vtkPlusOpenHapticsDevice()
+  :FrameNumber(-1)
+  , DeviceHandle(-1)
+  , DeviceName("PLUS")
+  , toolTransform(vtkSmartPointer<vtkTransform>::New())
+  , rotation(vtkSmartPointer<vtkTransform>::New())
+  , velMatrix(vtkSmartPointer<vtkMatrix4x4>::New())
+  , buttonMatrix(vtkSmartPointer<vtkMatrix4x4>::New())
+  , toolMatrix(vtkSmartPointer<vtkMatrix4x4>::New())
+{
+  this->RequirePortNameInDeviceSetConfiguration = true;
+  this->StartThreadForInternalUpdates = true;
+  this->AcquisitionRate = 20;
+
+  this->rasCorrection = vtkSmartPointer<vtkMatrix4x4>::New();
+  this->rasCorrection->Identity();
+  this->rasCorrection->SetElement(0, 0, -1);
+  this->rasCorrection->SetElement(1, 1, 0);
+  this->rasCorrection->SetElement(1, 2, 1);
+  this->rasCorrection->SetElement(2, 2, 0);
+  this->rasCorrection->SetElement(2, 1, 1);
+}
+
+vtkPlusOpenHapticsDevice::~vtkPlusOpenHapticsDevice(){}
+
+std::string vtkPlusOpenHapticsDevice::GetSdkVersion()
+{
+  std::stringstream ss;
+  ss << HD_VERSION_MAJOR_NUMBER << "." << HD_VERSION_MINOR_NUMBER << "." << HD_VERSION_BUILD_NUMBER;
+  return ss.str();
+}
+
+PlusStatus vtkPlusOpenHapticsDevice::Probe()
+{
+  if (this->Connected)
+  {
+    LOG_ERROR("vtkPlusOpenHapticsDevice::Probe should not be called while the device is already initialized");
+    return PLUS_FAIL;
+  }
+  if (this->InternalConnect())
+  {
+    this->InternalDisconnect();
+    LOG_INFO("vtkPlusOpenHapticsDevice::Probe found OpenHaptics Device");
+    return PLUS_SUCCESS;
+  }
+  else
+  {
+    LOG_ERROR("vtkPlusOpenHapticsDevice::Probe failed to connect to device");
+    return PLUS_FAIL;
+  }
+
+}
+
+PlusStatus vtkPlusOpenHapticsDevice::InternalUpdate()
+{
+  //schedule the internal callback - must be done each time device is polled
+  //the scheduler is intended to manage multiple clients requesting data from the same device
+  hdScheduleSynchronous(positionCallback, this, HD_MAX_SCHEDULER_PRIORITY);
+
+  return PLUS_SUCCESS;
+}
+
+PlusStatus vtkPlusOpenHapticsDevice::InternalConnect()
+{
+
+  HDErrorInfo errorFlush;
+  while (HD_DEVICE_ERROR(errorFlush = hdGetError())) {}
+
+  this->DeviceHandle = hdInitDevice(this->DeviceName.c_str());
+
+  HDErrorInfo error;
+  if (HD_DEVICE_ERROR(error = hdGetError()))
+  {
+    LOG_ERROR( "Failed to initialize Phantom Omni " << this->DeviceName);
+    this->DeviceHandle = -1;
+    return PLUS_FAIL;
+  }
+
+  // Enable forces
+  hdEnable(HD_FORCE_OUTPUT);
+  hdEnable(HD_FORCE_RAMPING);
+
+  LOG_DEBUG("Phantom initialized: " << this->DeviceName)
+
+  hdStartScheduler();
+
+  return PLUS_SUCCESS;
+}
+
+PlusStatus vtkPlusOpenHapticsDevice::InternalDisconnect()
+{
+  hdDisableDevice(DeviceHandle);
+  hdStopScheduler();
+  return PLUS_SUCCESS;
+}
+
+PlusStatus vtkPlusOpenHapticsDevice::ReadConfiguration(vtkXMLDataElement* rootConfigElement)
+{
+  XML_FIND_DEVICE_ELEMENT_REQUIRED_FOR_READING(deviceConfig, rootConfigElement);
+  XML_READ_CSTRING_ATTRIBUTE_REQUIRED(DeviceName,deviceConfig);

```

Comment:
Minor code formatting, space between argument and comma

See Astyle formatter (if using Visual Studio, available as an extension) and Astyle formatting rules [here](https://github.com/PlusToolkit/PlusBuild/blob/master/Docs/BuildInstructionsWindows.md).

Reply:
Also, since this is a std::string, you can use the XML_READ_STRING instead of CSTRING